File: CRYPTANALYSIS.py
def encrypt_caesar(text, shift):
    result = ""
    for ch in text:
        if ch.islower():
            result += chr((ord(ch) - ord('a') + shift) % 26 + ord('a'))
        elif ch.isupper():
            result += chr((ord(ch) - ord('A') + shift) % 26 + ord('A'))
        else:
            result += ch
    return result

def decrypt_all_shifts(cipher_text, original_text):
    for i in range(26):
        decrypted = ""
        for ch in cipher_text:
            if ch == " ":
                decrypted += " "
            elif ch.isupper():
                decrypted += chr((ord(ch) - ord('A') - i + 26) % 26 + ord('A'))
            elif ch.islower():
                decrypted += chr((ord(ch) - ord('a') - i + 26) % 26 + ord('a'))
            else:
                decrypted += ch
        print(f"Shift {i}: {decrypted}")
        if decrypted == original_text:
            print(f"Correct shift is: {i}")
            return
    print("Correct shift not found")

# === Main Execution ===
original_text = input("Enter the plain text: ")
shift = int(input("Enter the desired shift: "))

encrypted_text = encrypt_caesar(original_text, shift)
print("Encrypted text:", encrypted_text)

decrypt_all_shifts(encrypted_text, original_text)

File: C++ codes/EXPT 2/CRYPTANALYSIS.cpp
#include<iostream>
using namespace std;
int main()
{
    //caeser cypher brute force cryptanalysis
    string s;
    cout<<"Enter the plain text: ";
    getline(cin,s);
    string q=s;
    cout<<"Enter the desired shift: ";
    int shift;
    cin>>shift;
    for(int i=0;i<s.length();i++)
    {
        if(s[i]>='a' && s[i]<='z')
        {
            s[i]=char((int(s[i]-'a')+shift)%26+'a');
        }
        else if(s[i]>='A' && s[i]<='Z')
        {
            s[i]=char((int(s[i]-'A')+shift)%26+'A');
        }
    }
    cout<<"Encrypted text: "<<s<<endl;
    for(int i=0;i<26;i++)
    {
        cout<<"Shift "<<i<<": ";
        string b;
        for(int j=0;j<s.length();j++)
        {
            if(s[j]==' ')
            {
                cout<<" ";
                continue;
            }
            if(isupper(s[j]))
            {
                b+=char(int(s[j]-65-i+26)%26+65);
                cout<<char(int(s[j]-65-i+26)%26+65);
            }
            else
            {
                b+=char(int(s[j]-97-i+26)%26+97);
                cout<<char(int(s[j]-97-i+26)%26+97);
            }
        }
        cout<<endl;
        if(b==q)
        {
            cout<<"Shift is: "<<i<<endl;
            return 0;
        }
    }
    cout<<"Shift not found"<<endl;
    return 0;
}